Lib/_markupbase.py

```diff
@@ -276,13 +276,14 @@ def _parse_doctype_attlist(self, i, declstartpos):
                 return -1
             if c == "(":
                 # an enumerated type; look for ')'
-                if ")" in rawdata[j:]:
-                    j = rawdata.find(")", j) + 1
+                _temp = rawdata.find(")", j)
+                if _temp != -1:
+                    j = _temp + 1
                 else:
                     return -1
-                while rawdata[j:j+1].isspace():
+                while rawdata[j].isspace():
```

Comment:
If `j` becomes `len(rawdata)`, this will now fail with `IndexError`. Previously the loop just stopped and it went into the if statement below.

Reply:
Ok, that makes sense.
It would require two checks to make this behave properly (j < len(rawdata) and rawdata[j].isspace()).
performance improvement for this is not worth the complexity